pkg/controllers/nodeclaim/lifecycle/controller.go

```diff
@@ -171,6 +171,21 @@ func (c *Controller) Reconcile(ctx context.Context, nodeClaim *v1.NodeClaim) (re
 
 //nolint:gocyclo
 func (c *Controller) finalize(ctx context.Context, nodeClaim *v1.NodeClaim) (reconcile.Result, error) {
+	// perform a no-op for whatever the status condition is currently set to
+	// so that we bump the observed generation to the latest and prevent the nodeclaim
+	// root status from entering an `Unknown` state
+	stored := nodeClaim.DeepCopy()
+	for _, condition := range nodeClaim.GetConditions() {
+		if condition.ObservedGeneration != nodeClaim.Generation {
+			nodeClaim.StatusConditions().Set(condition)
+		}
+	}
+	if !equality.Semantic.DeepEqual(stored, nodeClaim) {
+		if err := c.kubeClient.Status().Patch(ctx, nodeClaim, client.MergeFrom(stored)); err != nil {
```

Comment:
Can you make sure this is with the OptimisticLock? We're messing with the status condition so we need to make sure that we aren't overriding previous changes to status conditions

Reply:
Thought about it but is it needed here given that this nodeclaim is being deleted? 

The other lifecycle status conditions won't race because we enter finalize before the other lifecycle methods reconcile and the other status conditions that would be added include the ones from Disruption, Drift, Consolidation, and Consistency checks.

Either way, I'm fine adding it.

Reply:
I think it's safer to include it generally -- it's possible that we build something down the line that updates during the delete and we'll have this race because we forgot that we didn't optimistic lock here